cypress/integration/basic_tests.js

```diff
@@ -0,0 +1,62 @@
+/// <reference types="Cypress" />
+
+
+
+describe('Basic Login and Signup', function() {
+  beforeEach(function() {
+    // reset the database prior to every test
+    cy.resetDatabase()
+
+    // Create default admin user
+    cy.fixture('defaultUser').as('defaultUser').then(async () => {
+      cy.createUser(this.defaultUser)
+    })
+  })
+  it('Can successfully login with token', function() {
+    cy.login(this.defaultUser)
+    cy.visit('/')
+    cy.contains(this.defaultUser.username)
+  })
+  it('Successfully sign up with basic user', function() {
+    const randomUserName = `testUser${Date.now()}`
+    cy.visit('/')
+    cy.contains("Login").click()
+    cy.contains("Sign up").click()
+    cy.get('#username')
+      .type(randomUserName)
+    cy.get('#email')
+      .type(`${randomUserName}@example.com`)
+    cy.get('#password')
+      .type('examplePassword1234')
+    cy.contains("Sign up").click()
+    cy.get('.users-menu')
+    cy.contains('Recommended Sequences')
+  })
+  it('Successfully logs in and out with default user', function() {
+    const { username, password } = this.defaultUser
```

Comment:
How does carried-over state work between these, exactly? The previous test created a user, so it will be logged in at the end of it. This one depends on defaultUser user existing, but also depends on not being logged in as them, so some state is being carried over (database) but some state isn't (cookies?). Is it the `it()` that resets the cookies, or the `cy.visit`?